### PR TITLE
modify some typos when I  study sparse tensor

### PR DESCRIPTION
**Pull request by OccupyMars2030**
### PR types

Others 
### PR changes

Others 
### Describe

modify some typos

### CODE REVIEW

File: paddle/fluid/eager/grad_node_info.h [hunk 1]
```diff
@@ -253,14 +254,14 @@ class GradNodeBase {
...
-  std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>
+  std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>&
   GetGradientHookFuntions() {
     VLOG(6) << "GetGradientHookFuntions ";
     return gradient_hooks_;
   }
 
   void SetGradientHookFuntions(
-      std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>
+      std::map<int64_t, std::tuple<size_t, size_t, std::shared_ptr<TensorHook>>>&
```
Comment (OccupyMars2030) [inline paddle/fluid/eager/grad_node_info.h]: In my opinion, GetGradientHookFuntions  and  SetGradientHookFuntions  can use reference type  for passing value to reduce copy time.  And the only place where these two methods are used is here:
paddle\fluid\eager\general_grad.h
image

It seems ok to do the modification.

File: paddle/fluid/eager/accumulation/accumulation_node.h [hunk 2]
```diff
@@ -60,8 +60,7 @@ class GradNodeAccumulation : public GradNodeBase {
   void ApplyReduceHooks();
 
   std::shared_ptr<GradNodeBase> Copy() const override {
-    return std::shared_ptr<GradNodeAccumulation>(
-        new GradNodeAccumulation(nullptr));
+    return std::shared_ptr<GradNodeAccumulation>(nullptr);
```
Comment (OccupyMars2025) [inline paddle/fluid/eager/accumulation/accumulation_node.h:63]: https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr
image

File: paddle/fluid/pybind/eager.h [hunk 3]
```diff
@@ -21,7 +21,8 @@ namespace paddle {
 namespace pybind {
 
 typedef struct {
-  PyObject_HEAD paddle::experimental::Tensor tensor;
+  PyObject_HEAD 
+  paddle::experimental::Tensor tensor;
```
Comment (OccupyMars2030) [inline paddle/fluid/pybind/eager.h]: image
Include\object.h

Comment (OccupyMars2030) [reply paddle/fluid/pybind/eager.h]: It's better to put PyObject_HEAD  at a single line.

File: paddle/fluid/pybind/eager_functions.cc [hunk 4]
```diff
@@ -958,11 +959,11 @@ PyMethodDef variable_functions[] = {
      NULL},
     /**sparse functions**/
     {"sparse_coo_tensor",
-     (PyCFunction)(void (*)(void))eager_api_sparse_coo_tensor,
+     (PyCFunctionWithKeywords)(void (*)(void))eager_api_sparse_coo_tensor,
      METH_VARARGS | METH_KEYWORDS,
      NULL},
     {"sparse_csr_tensor",
-     (PyCFunction)(void (*)(void))eager_api_sparse_csr_tensor,
+     (PyCFunctionWithKeywords)(void (*)(void))eager_api_sparse_csr_tensor,
```
Comment (OccupyMars2030) [inline paddle/fluid/pybind/eager_functions.cc]: https://docs.python.org/3.10/c-api/structures.html?highlight=pycfunction#implementing-functions-and-methods   

image

According to the signature of eager_api_sparse_coo_tensor  and  eager_api_sparse_csr_tensor,  it seems more suitable  to  use  PyCFunctionWithKeywords

Comment (OccupyMars2030) [reply paddle/fluid/pybind/eager_functions.cc]: In CPython source code,  in  the file  Include\methodobject.h  

image

Comment (OccupyMars2030) [reply paddle/fluid/pybind/eager_functions.cc]: image

Comment (OccupyMars2030) [reply paddle/fluid/pybind/eager_functions.cc]: image

**Pull request closed without merging**